app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt

```diff
@@ -45,13 +46,20 @@ class NewsAdapter(private val dataSet: ArrayList<New>) :
 
     // Replace the contents of a view (invoked by the layout manager)
     override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
-
         // Get element from your dataset at this position and replace the
         // contents of the view with that element
         news[position].also {
+            viewHolder.id = it.id!!
             viewHolder.commenter.text = it.commenter
             viewHolder.comment.text = it.comment
             viewHolder.timeAgo.text = it.date.formatTime()
+            viewHolder.likeState.isChecked = fragment.toggleLikeButton(it.likes)
+        }
+        viewHolder.itemView.setOnClickListener {
+            fragment.goToNewDetail(viewHolder.id)
+        }
+        viewHolder.likeState.setOnClickListener {
+            fragment.updateLike(viewHolder.id)
```

Comment:
Remember to name the listener's methods as events, like `onNewDetailClicked()` and `onLikeClicked()`